TTHAnalysis/python/plotter/ttH-multilepton/lepchoice-ttH-FO.txt

```diff
@@ -2,10 +2,10 @@ LepGood1_isLepTight\b : LepGood_isLepTight_Recl[iLepFO_Recl[0]]; AlsoData
 LepGood2_isLepTight\b : LepGood_isLepTight_Recl[iLepFO_Recl[1]]; AlsoData
 LepGood3_isLepTight\b : LepGood_isLepTight_Recl[iLepFO_Recl[2]]; AlsoData
 LepGood4_isLepTight\b : LepGood_isLepTight_Recl[iLepFO_Recl[3]]; AlsoData
-LepGood1_isMatchRightCharge\b : (LepGood_mcMatchId[iLepFO_Recl[0]]!=0 && LepGood_pdgId[iLepFO_Recl[0]]==LepGood_mcMatchPdgId[iLepFO_Recl[0]]); AlsoData
-LepGood2_isMatchRightCharge\b : (LepGood_mcMatchId[iLepFO_Recl[1]]!=0 && LepGood_pdgId[iLepFO_Recl[1]]==LepGood_mcMatchPdgId[iLepFO_Recl[1]]); AlsoData
-LepGood3_isMatchRightCharge\b : (LepGood_mcMatchId[iLepFO_Recl[2]]!=0 && LepGood_pdgId[iLepFO_Recl[2]]==LepGood_mcMatchPdgId[iLepFO_Recl[2]]); AlsoData
-LepGood4_isMatchRightCharge\b : (LepGood_mcMatchId[iLepFO_Recl[3]]!=0 && LepGood_pdgId[iLepFO_Recl[3]]==LepGood_mcMatchPdgId[iLepFO_Recl[3]]); AlsoData
+LepGood1_isMatchRightCharge\b : (LepGood_genPartFlav[iLepFO_Recl[0]]==1 || LepGood_genPartFlav[iLepFO_Recl[0]]==15) && (GenPart_pdgId[LepGood_genPartIdx[iLepFO_Recl[0]]]*LepGood_pdgId[iLepFO_Recl[0]]>0); AlsoData
+LepGood2_isMatchRightCharge\b : (LepGood_genPartFlav[iLepFO_Recl[1]]==1 || LepGood_genPartFlav[iLepFO_Recl[1]]==15) && (GenPart_pdgId[LepGood_genPartIdx[iLepFO_Recl[1]]]*LepGood_pdgId[iLepFO_Recl[1]]>0); AlsoData
+LepGood3_isMatchRightCharge\b : (LepGood_genPartFlav[iLepFO_Recl[2]]==1 || LepGood_genPartFlav[iLepFO_Recl[2]]==15) && (GenPart_pdgId[LepGood_genPartIdx[iLepFO_Recl[2]]]*LepGood_pdgId[iLepFO_Recl[2]]>0); AlsoData
+LepGood4_isMatchRightCharge\b : (LepGood_genPartFlav[iLepFO_Recl[3]]==1 || LepGood_genPartFlav[iLepFO_Recl[3]]==15) && (GenPart_pdgId[LepGood_genPartIdx[iLepFO_Recl[3]]]*LepGood_pdgId[iLepFO_Recl[3]]>0); AlsoData
```

Comment:
we should check it it would be worth making at a new branch in the recleaner for performance reasons